src/Psalm/Internal/Analyzer/MethodAnalyzer.php

```diff
@@ -206,6 +206,26 @@ public static function checkMethodNotDeprecatedOrInternal(
             }
         }
 
+        if ($storage->psalmInternal
+            && $context->self
+            && !$context->collect_initializations
+            && !$context->collect_mutations
+        ) {
+            if (strpos($context->self, trim($storage->psalmInternal, '\\') . '\\') !== 0) {
```

Comment:
I've made six copies of essentially same function built around `strpos` to check if one namespace is inside another. I'd like to extract it to a function, but I'm not sure if there's a good place in the codebase to put it. Same applies to the pre-existing code used to check if two namespaces are unrelated.

Reply:
And would be good to also make the check case-insensitive.